### PR TITLE
Memory lecturer interface

### PR DESCRIPTION
**Pull request by PatrickHoltz**
closes https://github.com/Gamify-IT/issues/issues/451
Card pairs can now be configured with their type and content.

Closes Gamify-IT/issues#480

### CODE REVIEW

**Comment by maexled** [comment]
@madebyTimo can you just have a look over the updated nginx and docker compose files?

File: src/components/EditMinigameModals/EditMemoryConfigurationModal.vue [hunk 1]
```diff
@@ -0,0 +1,414 @@
...
+import { ITask } from "@/ts/models/overworld-models";
+import ImportExportConfiguration from "@/components/ImportExportConfiguration.vue";
+import {
+  IMemoryCardPair,
+  MemoryCard,
+  MemoryCardPair,
+  MemoryCardType,
+  MemoryConfiguration,
+} from "@/ts/models/memory-models";
+import { config } from "process";
```
Comment (maexled) [inline src/components/EditMinigameModals/EditMemoryConfigurationModal.vue:23]:
```suggestion
```
unused

File: src/components/EditMinigameModals/EditMemoryConfigurationModal.vue [hunk 2]
```diff
@@ -0,0 +1,414 @@
+<script setup lang="ts">
+import { saveAs } from "file-saver";
+import { arrayOf, defaultValue, int, object, oneOf, string } from "checkeasy";
+import { importConfiguration } from "@/ts/import-configuration";
+import { Ref, defineEmits, defineProps, onMounted, ref, watch } from "vue";
```
Comment (maexled) [inline src/components/EditMinigameModals/EditMemoryConfigurationModal.vue:5]:
```suggestion
import { arrayOf, object, oneOf, string } from "checkeasy";
import { importConfiguration } from "@/ts/import-configuration";
import { Ref, defineEmits, defineProps, ref, watch } from "vue";
```
remove unused imports

File: src/components/EditMinigameModals/EditMemoryConfigurationModal.vue [hunk 3]
```diff
@@ -0,0 +1,414 @@
...
+const showEditModal = ref(false);
+const inEditModal = ref(false);
+
+const card1Type = ref();
+const card1Content = ref();
+const card2Type = ref();
+const card2Content = ref();
+const editObject = ref();
+
+let memoryCardTypeValues = Object.values(MemoryCardType);
```
Comment (maexled) [inline src/components/EditMinigameModals/EditMemoryConfigurationModal.vue:68]:
```suggestion
const memoryCardTypeValues = Object.values(MemoryCardType);
```

File: src/components/EditMinigameModals/EditMemoryConfigurationModal.vue [hunk 4]
```diff
@@ -0,0 +1,414 @@
...
+  (e: "closedModal"): void;
+}>();
+
+function checkFormValidity(): boolean {
+  return form.value.checkValidity();
+}
+
+function initializePairs() {
+  cardPairs.value = [];
+  for (let index = 0; index < 6; index++) {
```
Comment (maexled) [inline src/components/EditMinigameModals/EditMemoryConfigurationModal.vue:115]: 6 seems a magic number here. Declare a new variable that indicates that 6 is the default amount of pairs.

File: src/components/EditMinigameModals/EditMemoryConfigurationModal.vue [hunk 5]
```diff
@@ -0,0 +1,414 @@
...
+            <span
+              v-if="
+                data.item.card1.type == MemoryCardType.TEXT ||
+                data.item.card1.type == MemoryCardType.MARKDOWN
+              "
+            >
+              {{ data.item.card1.content.slice(0, 10) }}
+            </span>
+            <span v-if="data.item.card1.content.length > 10">...</span>
+          </template>
```
Comment (maexled) [inline src/components/EditMinigameModals/EditMemoryConfigurationModal.vue:325]: The card type and card content should be in a different style to make visually clear these are two different things. Maybe make the card type bold and the content normal? Or make the content a smaller font. Just ideas.

**Review by maexled — COMMENTED** [review]
Good work! The functionality works fine.

**Review by maexled — APPROVED** [review]

**Review by C0okiedestroyer — APPROVED** [review]

**Pull request merged**